packages/react-heartwood-components/src/components/Text/Text.js

```diff
@@ -1,16 +1,69 @@
 // @flow
-import React from 'react'
+import React, { Fragment } from 'react'
 import cx from 'classnames'
 
+import Button from '../Button/Button'
+import TextStyle from '../TextStyle/TextStyle'
+
 import type { Node } from 'react'
 
+// Components available for templating
+
+const TextComponentKey = {
+	style: TextStyle,
+	button: Button
+}
+
+// Allows basic templating functionality on text strings
+
+const TemplateEngine = (text = '', context = {}) => {
+	let re = /{{([^}}]+)?}}/g,
+		children = [],
+		cursor = 0,
+		match
+
+	let add = function(line, js) {
+		if (line !== '') {
+			children.push({
+				props: { element: 'span', children: line.replace(/"/g, '\\"') }
+			})
+		}
+		if (context[js]) {
+			children.push(context[js])
+		}
+	}
+	while ((match = re.exec(text))) {
```

Comment:
Standard warning to be absolutely sure there's no way this can result in an endless loop 😬

Reply:
(Seems fine at a glance, though)

Reply:
Updated to store matches first and then iterating over them while still calling exec in order to ensure a finite loop. Little weird but less scary...